Program security - Owner check lesson updated

0xCipherCoder opened the pull request:
### Problem

### Summary of Changes

- Updated code snippets with the latest anchor version
- Fixed content,
- Fixed grammar and styling
- Fixed as per guidelines

Fixes #
Unboxed PRs
Starter - https://github.com/Unboxed-Software/solana-owner-checks/pull/5
Solution - https://github.com/Unboxed-Software/solana-owner-checks/pull/6

<!-- 
Note: The maintainers of this repo may make editorial changes as needed without creating comments.
Please ensure you have "Allow edits by maintainers" setting on this PR enabled to help speed up the review process. Thanks :)
-->

mikemaccana commented on content/courses/program-security/owner-checks.md:
```diff
@@ -3,10 +3,10 @@ title: Owner Checks
     checks
-  - Implement owner checks using long-form Rust
-  - Use Anchor’s `Account<'info, T>` wrapper and an account type to automate
+  - Implement owner checks using native Rust
+  - Use Anchor's `Account<'info, T>` wrapper and an account type to automate
```
Just swap these two (and do it everywhere, Anchor is default for onchain programs)

mikemaccana commented on content/courses/program-security/owner-checks.md:
```suggestion
represents the **program** that owns the account. Owner checks ensure that this
```

mikemaccana commented on content/courses/program-security/owner-checks.md:
```diff
@@ -227,19 +202,18 @@ pub struct AdminConfig {
-derived from a different program. You can use the `seeds` and `bump` constraints
-and define the `owner` to properly derive and verify the address of the account
-passed in.
+In addition to the `Account` type, you can use the `owner` constraint to specify
```
Add a link to https://www.anchor-lang.com/docs/account-constraints (unfortunately Anchor doesn't support, er, anchors)

0xCipherCoder replied:
Updated anchor link as per comment.

mikemaccana reviewed (approved):
Fine! Have some small changes,  please do them, but happy to mark this as the winner for owner-checks.

0xCipherCoder commented:
> Fine! Have some small suggested changes, please do them, but happy to mark this as the winner for owner-checks.

Thanks! @mikemaccana  I have updated the content per the comment and resolved all of them. Also refactored tests in the starter and solution branches and updated the lesson content accordingly. Please review and let me know if any additional feedback.

mikemaccana commented:
@0xCipherCoder can you change repo references to https://github.com/solana-developers/owner-checks/?

0xCipherCoder commented:
> @0xCipherCoder can you change repo references to https://github.com/solana-developers/owner-checks/?

@mikemaccana I have updated repo references. Please check.

mikemaccana commented:
Thanks @0xCipherCoder !
